src/Map.ts

```diff
@@ -175,9 +175,10 @@ export class Map extends maplibregl.Map {
   private isTerrainEnabled = false;
   private terrainExaggeration = 1;
   private primaryLanguage: LanguageString;
-  private secondaryLanguage?: LanguageString;
   private terrainGrowing = false;
   private terrainFlattening = false;
+  private forceLanguageUpdate: boolean;
+  private languageAlwaysBeenStyle: boolean;
```

Comment:
This name confused me, but maybe it can be `primaryLanguageIsStyle`? Just spit balling. Maybe it's fine as is.

Reply:
Yeah i was not sure how to call it but `languageAlwaysBeenStyle` is true if the language has always been the one defined in the style (meaning the map ash been instanciated with `Language.STYLE` or with nothing because this is the default anyway. Then if at any moment the language of the map is changed to something else (eg. `Language.ENGLISH`) then `languageAlwaysBeenStyle` becomes `false` and will never be turned to `true` ever, even if later the language is turned to `Language.STYLE` again.

The role of this is to prevent unnecessary language update in the case of the language being just as the style is delivered.